modules/dnn/src/cuda4dnn/primitives/layer_norm.hpp

```diff
@@ -43,10 +52,25 @@ namespace cv { namespace dnn { namespace cuda4dnn {
                      const std::vector<cv::Ptr<BackendWrapper>>& outputs,
                      csl::Workspace& workspace) override {
             auto input_wrapper = inputs[0].dynamicCast<wrapper_type>();
-            auto scale_wrapper = inputs[1].dynamicCast<wrapper_type>();
-
             auto input = input_wrapper->getView();
-            auto scale = scale_wrapper->getView();
+
+            csl::TensorView<T> scale;
+            if (input_scale_tensor.empty()) {
+                auto scale_wrapper = inputs[1].dynamicCast<wrapper_type>();
+                scale = scale_wrapper->getView();
+            } else {
+                scale = csl::TensorView<T>(input_scale_tensor);
+            }
+
+            csl::TensorView<T> bias;
+            if (input_bias_tensor.empty()) {
+                if (inputs.size() == 3) {
```

Comment:
m.b. `inputs.size() >= 3` is safer here.

Reply:
Resolved